presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2802,6 +2802,306 @@ public void testCollectColumnStatisticsOnInsert()
+    }
+
+    @Test
+    public void testInvalidAnalyzeUnpartitionedTable()
+    {
+        String tableName = "test_invalid_analyze_unpartitioned_table";
+
+        // Test table does not exist
+        assertQueryFails("ANALYZE " + tableName, ".*Table.*does not exist.*");
+
+        createUnpartitionedTableForAnalyzeTest(tableName);
+
+        // Test partition properties on unpartitioned table
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[])", tableName), ".*Only partitioned table can be analyzed with a partition list.*");
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p1']])", tableName), ".*Only partitioned table can be analyzed with a partition list.*");
+
+        // Drop the partitioned test table
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testAnalyzePartitionedTable()
+    {
+        String tableName = "test_analyze_partitioned_table";
+        createPartitionedTableForAnalyzeTest(tableName);
+
+        // No column stats before running analyze
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s)", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, null, null, null, null, null), " +
+                        "('c_bigint', null, null, null, null, null, null), " +
+                        "('c_double', null, null, null, null, null, null), " +
+                        "('c_timestamp', null, null, null, null, null, null), " +
+                        "('c_varchar', null, null, null, null, null, null), " +
+                        "('c_varbinary', null, null, null, null, null, null), " +
+                        "('p_varchar', 24.0E0, 6.0E0, 0.25E0, null, null, null), " +
+                        "(null, null, null, null, 16.0E0, null, null)");
+
+        // No column stats after running an empty analyze
+        assertUpdate(format("ANALYZE %s WITH (partitions = ARRAY[])", tableName));
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s)", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, null, null, null, null, null), " +
+                        "('c_bigint', null, null, null, null, null, null), " +
+                        "('c_double', null, null, null, null, null, null), " +
+                        "('c_timestamp', null, null, null, null, null, null), " +
+                        "('c_varchar', null, null, null, null, null, null), " +
+                        "('c_varbinary', null, null, null, null, null, null), " +
+                        "('p_varchar', 24.0E0, 6.0E0, 0.25E0, null, null, null), " +
+                        "(null, null, null, null, 16.0E0, null, null)");
+
+        // Run analyze on 3 partitions including the null partition
+        assertUpdate(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p1'], ARRAY['p2'], ARRAY[NULL]])", tableName), 12);
```

Comment:
Add a test for a table partitioned by timestamp (you may just add another partitioning column to what you have now).
In particular, I looking into covering interaction between 
* (yours) `com.facebook.presto.hive.HivePartitionManager#getPartitions(com.facebook.presto.hive.metastore.SemiTransactionalHiveMetastore, com.facebook.presto.spi.ConnectorTableHandle, java.util.List<java.util.List<java.lang.String>>)`
* and the following:
 `com.facebook.presto.hive.HivePartitionManager#extractPartitionKeyValues`, `com.facebook.presto.hive.HiveUtil#parsePartitionValue`